database/sql/V072__auction_solution_orders.sql

```diff
@@ -0,0 +1,36 @@
+-- Table to store all proposed solutions for an auction, received from solvers during competition time.
+-- A single auction can have multiple solutions, and each solution can contain multiple orders.
+-- Order is not allowed to be part of more than one solution within the same auction.
+
+-- `solver` + `solution_id` are unique within an auction.
+
+-- `deadline` is used to determine if the order is part of a winning solution.
+
+CREATE TABLE auction_solution_orders (
+   auction_id bigint NOT NULL,
+   solver bytea NOT NULL,
+   solution_id string NOT NULL,
+   order_uid bytea NOT NULL,
+   -- The block number until which the order should be settled.
+   -- Not NULL for winning orders.
+   deadline bigint,
+
+   -- Order details
+   sell_token bytea NOT NULL,
+   buy_token bytea NOT NULL,
+   limit_sell numeric(78,0) NOT NULL,
+   limit_buy numeric(78,0) NOT NULL,
+   side OrderKind NOT NULL,
+   -- Uniform clearing price of the sell token
+   sell_token_price numeric(78,0) NOT NULL,
+   -- Uniform clearing price of the buy token
+   buy_token_price numeric(78,0) NOT NULL,
```

Comment:
If we are actually talking about UCP this would be a property on a solution.
If we no longer want to be married to the idea to 1 price per block having this on the order itself would make sense to me.

Reply:
Putting it on the order seems like more future proof, as you said, if we drop UCP requirement.

If we want to enforce UCP right now, I think we can do validation on inserting/fetching.

Reply:
I meant it's more general meaning orders in a same solution don't have to have same uniform clearing price for common tokens. But again, if we need this guarantee, we can enforce it at insert/fetch time

Reply:
I moved out the UCPs as JIT order won't have them, or better said, specifying UCPs for JIT orders will not be required by the driver API.

Reply:
Not commenting on the table setup but on requiring clearing prices: why do we not require uniform clearing prices for jit orders? Is there a discussion somewhere on what types of orders should exist and how the interface between driver and autopilot is designed? (Imo, having the old concept of liquidity orders  around creates complications which require special handling in the mechanism. Intended use cases are not worth this effort.)

Reply:
There is some discussion around liquidity orders encoding for the settlement contract: https://github.com/cowprotocol/services/pull/3032#issue-2559281410

There, we mention that non-surplus-capturing JIT orders are not supposed to encode UCP, but only custom prices. Therefore, I assumed we want to keep the same requirements for both encoding and for `/solve` response:

1. User orders required both uniform and custom prices
2. Surplus capturing JIT orders required both uniform and custom prices
3. Non surplus capturing JIT orders required only custom prices